cmd/oras/root/pull.go

```diff
@@ -213,20 +222,24 @@ func doPull(ctx context.Context, src oras.ReadOnlyTarget, dst oras.GraphTarget,
 					config.Annotations[ocispec.AnnotationTitle] = configPath
 				}
 			})
-			nodes = append(nodes, *config)
+			if config.Size != ocispec.DescriptorEmptyJSON.Size || config.Digest != ocispec.DescriptorEmptyJSON.Digest || config.Annotations[ocispec.AnnotationTitle] != "" {
```

Comment:
[`content.Equal()`](https://pkg.go.dev/oras.land/oras-go/v2/content#Equal) is a better method to compare two descriptors.

Reply:
No, it cannot be used here since the comparison should skip `mediaType`